src/providers/json-rpc-provider.ts

```diff
@@ -348,6 +348,10 @@ export class JsonRpcProvider extends Provider {
     async sendJsonRpc<T>(method: string, params: object): Promise<T> {
         const response = await exponentialBackoff(REQUEST_RETRY_WAIT, REQUEST_RETRY_NUMBER, REQUEST_RETRY_WAIT_BACKOFF, async () => {
             try {
+                if ('blockId' in params) {
```

Comment:
seems like this logic should be in `query`

also probably can be smth like:
```
const { blockId: block_id, ...otherParams } = params;
params = { ...otherParams, block_id };
```

normally you won't use object fields as dictionary